services/replicator/replicator.go

```diff
@@ -219,6 +219,10 @@ func (r *Replicator) OpenReplicationReadStreamHandler(w http.ResponseWriter, req
 		}).Error("Can't open store host read stream")
 		r.m3Client.IncCounter(metrics.OpenReplicationReadScope, metrics.ReplicatorCreateOutStreamFailure)
 		r.m3Client.IncCounter(metrics.OpenReplicationReadScope, metrics.ReplicatorFailures)
+
+		// Must close the connection on server side because closing on client side doesn't actually close the
+		// underlying TCP connection
+		inStream.Done()
```

Comment:
Probably better do a "defer inStream.Done()" immediately after the open and if-error block.